Code/Framework/AzCore/AzCore/Name/NameDictionary.h

```diff
@@ -135,5 +139,11 @@ namespace AZ
         //! so we keep track of them here to ensure their name data gets correctly cleaned up
         //! when this dictionary is shut down.
         Name m_deferredHead;
+
+        //! Set the maximum number of hash slots to 2^32
+        //! hash values will be mapped between [0, m_maxHashSlots)
+        //! Can only be configured at construction time and cannot change
+        //! value cannot be 0
+        const AZ::u64 m_maxHashSlots{ static_cast<AZ::u64>(AZStd::numeric_limits<Name::Hash>::max()) + 1 };
```

Comment:
Due to the constructors above, it doesn't seem like this default value would ever be used.

Reply:
This is here to allow for future NameDictionary constructors to added without leaving the value uninitialized